Dragonflight/PriestDiscipline.lua

```diff
@@ -555,40 +606,96 @@ spec:RegisterAbilities( {
 
     mind_blast = {
         id = 8092,
-        cast = 1.5,
+        cast = function() return 1.5 * haste end,
         charges = function()
             if talent.dark_indulgence.enabled then return 2 end
         end,
         cooldown = 9,
         recharge = function()
-            if talent.dark_indulgence.enabled then return 9 end
+            if talent.dark_indulgence.enabled then return 9*haste end
         end,
+        hasteCD = true,
         gcd = "spell",
         school = "shadow",
         damage = 1,
 
-        spend = 0,
+        spend = function() return talent.dark_indulgence.enabled and 0.0015 or 0.0025 end,
         spendType = "mana",
 
         startsCombat = true,
         texture = 136224,
 
         handler = function ()
             if talent.harsh_discipline.enabled then
-                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 9 then
+                addStack( "harsh_discipline" )
+                if buff.harsh_discipline.up and buff.harsh_discipline.stack == 8/talent.harsh_discipline.rank then
```

Comment:
`if buff.harsh_discipline.up and buff.harsh_discipline.stack == buff.harsh_discipline.max_stack - 1 then`